src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -306,33 +405,78 @@ private Ref getRef(String jiraIssueKey) {
 	}
 
 	private List<Ref> getAllRefs() {
+		return getRefs(ListBranchCommand.ListMode.ALL);
+	}
+
+	private List<Ref> getOnlyRemoteRefs() {
+		return getRefs(ListBranchCommand.ListMode.REMOTE);
+	}
+
+	private List<Ref> getRefs(ListBranchCommand.ListMode listMode) {
 		List<Ref> refs = new ArrayList<Ref>();
 		try {
-			refs = git.branchList().setListMode(ListBranchCommand.ListMode.ALL).call();
+			refs = git.branchList().setListMode(listMode).call();
 		} catch (GitAPIException e) {
-			LOGGER.error("Git could not get all references. Message: " + e.getMessage());
+			LOGGER.error("Git could not get references. Message: " + e.getMessage());
 		}
 		return refs;
 	}
 
+	private List<RevCommit> getCommitsFromDefaultBranch() {
+		return getCommits(defaultBranch, true);
+	}
+
 	private List<RevCommit> getCommits(Ref branch) {
+		return getCommits(branch, false);
+	}
+
+	private List<RevCommit> getCommits(Ref branch, boolean isDefaultBranch) {
 		List<RevCommit> commits = new ArrayList<RevCommit>();
 		if (branch == null) {
 			return commits;
 		}
+
+		File directory;
+		String branchNameComponents[] = branch.getName().split("/");
+		String branchShortName = branchNameComponents[branchNameComponents.length-1];
+		boolean canReleaseRepoDirectory = false;
+
+		if (isDefaultBranch) {
+			directory = new File(fsManager.getDefaultBranchPath());
+		}
+		else {
+			canReleaseRepoDirectory =! fsManager.isBranchDirectoryInUse(branchShortName);
+			directory = new File(fsManager.prepareBranchDirectory(branchShortName));
+		}
 		try {
-			git.checkout().setName(branch.getName()).call();
+			git.close();
+			git=git.open(directory);
+			git.checkout().setName(branchShortName).call();
 			Iterable<RevCommit> iterable = git.log().call();
 			for (RevCommit commit : iterable) {
 				commits.add(commit);
 			}
-		} catch (GitAPIException e) {
-			LOGGER.error(
-					"Git could not get commits for the branch: " + branch.getName() + " Message: " + e.getMessage());
+		} catch (IOException | GitAPIException e) {
+			LOGGER.error("Git could not get commits for the branch: "
+					+ branch.getName() + " Message: " + e.getMessage());
+		}
+		if (canReleaseRepoDirectory ) {
+			fsManager.releaseBranchDirectoryNameToTemp(branchShortName);
 		}
+		switchGitClientBackToDefaultDirectory();
 		return commits;
 	}
 
+	private void switchGitClientBackToDefaultDirectory() {
+		File directory = new File(fsManager.getDefaultBranchPath());
+		try {
+			git.close();
+			git=git.open(directory);
```

Comment:
We should have a guideline on how to format our code. I always insert spaces befor and after an equal sign "=" like above. @lw2011 and @dekome , do you agree on inserting more spaces?